apps/designer/app/shared/dom-utils/on-iframe-load.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Resolve a promise when an iframe has readyState complete.
+ * Checks periodically for the readyState to be complete, because somttimes the load event may simply not fire for various reasons.
+ */
+export const onIframeLoad = (iframe: HTMLIFrameElement | null) =>
+  new Promise((resolve) => {
+    if (iframe === null) return;
+    const check = () => {
+      const document = iframe.contentDocument || iframe.contentWindow?.document;
+      if (document?.readyState === "complete") {
```

Comment:
this is a great point actually, will rewrite to polling now, because supporting any remote domain is a very important use case and I totally forgot that we don't have access to document there